util/gpool/spmc/spmcpool.go

```diff
@@ -297,6 +300,11 @@ func (p *Pool[T, U, C, CT, TF]) AddProduceBySlice(producer func() ([]T, error),
 					Task: task,
 				}
 				inputCh <- task
```

Comment:
Maybe we will block it here after we close(`productCloseCh`). e.g. We get 3 tasks when putting the second task to `inputCh`, the `productCloseCh` is closed and the consumer is exit

Reply:
fix